src/lib/wallet/index.ts

```diff
@@ -77,10 +78,15 @@ export const getWalletConnection = ({
   return undefined;
 };
 
-const getWalletErrorType = ({ error }: { error: Error }) => {
-  const { message } = error;
+const getWalletErrorType = ({ error }: { error: DydxError }) => {
+  const { message, code } = error;
   const messageLower = message.toLowerCase();
 
+  // Metamask - already pending request
+  if (code === MetamaskErrorCodes.RESOURCE_UNAVAILABLE) {
+    return WalletErrorType.PendingMetamaskRequest;
+  }
```

Comment:
Is it possible for other wallets to emit the same error code?

Reply:
yes! it's actually a good thing (i think).
phantom shares the same error codes 
https://docs.phantom.app/solana/errors